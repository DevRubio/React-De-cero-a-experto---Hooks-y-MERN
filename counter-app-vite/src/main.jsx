import React from 'react'
import ReactDOM from "react-dom/client"
import {FirstApp} from './FirstApp'
import './styles.css'
import { CounterApp } from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title='Hola, Como estan?'/>
        <CounterApp value={12}/>        
    </React.StrictMode>
)