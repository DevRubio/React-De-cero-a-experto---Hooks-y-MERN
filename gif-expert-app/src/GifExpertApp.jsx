import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch','Dragon Ball']);

  const onAddCategory = (newCategory) =>{
    if(categories.includes(newCategory)) return;
    //setCategories([...categories, newCategory])
    setCategories([newCategory, ...categories])
    
  }

  return (
    <>
      <h1>GifExpertApp</h1>

      {/*Listado de Gifs */}
      <AddCategory 
        //setCategories={setCategories}
        onNewCategory={ (value) => onAddCategory(value) }
      />
      

      <ol>
        {categories.map((category)=>(
          <li key={category}>{category}</li>
        ))}
      </ol>
    </>
  );
};
