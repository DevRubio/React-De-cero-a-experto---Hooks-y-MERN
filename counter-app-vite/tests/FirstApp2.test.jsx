import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp/>', () => { 

  const title = 'Hola, Soy Goku'
  const subtitle= 'Soy un Subtitle'

  test('debe de hacer match con el snapshop', () => { 
    const {container} = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
   })
   
   test('Debe mostrar el mensaje "Hola. Soy Goku"', () => { 
      render(<FirstApp title={title} />)
      expect(screen.getByText(title)).toBeTruthy()
    })

    test('Debe mostrar el titulo en un H1', () => { 
      render(<FirstApp title={title} />)
      expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title)
     })

     test('Debe mostrar el subtitulo enviado por props', () => { 
      render(
        <FirstApp
          title={title}
          subtitle={subtitle}
        />
      )
      expect(screen.getAllByText(subtitle).length).toBe(1)
      })



 })