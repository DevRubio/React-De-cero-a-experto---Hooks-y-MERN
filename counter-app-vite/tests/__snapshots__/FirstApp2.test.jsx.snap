// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshop 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p />
  <p />
</div>
`;
